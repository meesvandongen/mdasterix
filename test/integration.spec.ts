import { describe, it, expect } from "bun:test";
import { flowContentBuilder } from "../src/index.js";

describe("integration", () => {
  it("works", () => {
    const value = flowContentBuilder()
      .heading({ depth: 1 }, (phrasingBuilder) => {
        phrasingBuilder.text({ value: "Heading" });
      })
      .paragraph({}, (phrasingBuilder) => {
        phrasingBuilder.text({ value: "Paragraph" });
      })
      .table(
        {
          align: ["left", "right"],
        },
        (tableContentBuilder) => {
          [1, 2, 3].forEach((i) => {
            tableContentBuilder.tableRow({}, (rowContentBuilder) => {
              rowContentBuilder
                .tableCell({}, (phrasingBuilder) => {
                  phrasingBuilder.text({ value: `${i}-1` });
                })
                .tableCell({}, (phrasingBuilder) => {
                  phrasingBuilder.text({ value: `${i}-2` });
                });
            });
          });
        },
      )
      .blockquote({}, (flowContentBuilder) => {
        flowContentBuilder.paragraph({}, (phrasingBuilder) => {
          phrasingBuilder.text({ value: "Blockquote" });
        });
      })
      .build();

    expect(value).toMatchSnapshot();
  });
});
