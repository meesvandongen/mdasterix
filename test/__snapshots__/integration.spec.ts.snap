// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`integration works 1`] = `
{
  "children": [
    {
      "children": [
        {
          "type": "text",
          "value": "Heading",
        },
      ],
      "depth": 1,
      "type": "heading",
    },
    {
      "children": [
        {
          "type": "text",
          "value": "Paragraph",
        },
      ],
      "type": "paragraph",
    },
    {
      "align": [
        "left",
        "right",
      ],
      "children": [
        {
          "children": [
            {
              "children": [
                {
                  "type": "text",
                  "value": "1-1",
                },
              ],
              "type": "tableCell",
            },
            {
              "children": [
                {
                  "type": "text",
                  "value": "1-2",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        {
          "children": [
            {
              "children": [
                {
                  "type": "text",
                  "value": "2-1",
                },
              ],
              "type": "tableCell",
            },
            {
              "children": [
                {
                  "type": "text",
                  "value": "2-2",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        {
          "children": [
            {
              "children": [
                {
                  "type": "text",
                  "value": "3-1",
                },
              ],
              "type": "tableCell",
            },
            {
              "children": [
                {
                  "type": "text",
                  "value": "3-2",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
    {
      "children": [
        {
          "children": [
            {
              "type": "text",
              "value": "Blockquote",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "root",
}
`;
